src/unit_tests/ctx/add_to_library.rs

```diff
@@ -143,3 +144,137 @@ fn actionctx_addtolibrary() {
         "datastorePut request has been sent"
     );
 }
+
+#[test]
+fn actionctx_addtolibrary_already_added() {
+    #[derive(Model, Debug, Default)]
+    struct Model {
+        ctx: Ctx<Env>,
+    }
+    fn fetch_handler(request: Request) -> EnvFuture<Box<dyn Any>> {
+        match request {
+            Request {
+                url, method, body, ..
+            } if url == "https://api.strem.io/api/datastorePut"
+                && method == "POST"
+                && body == "{\"authKey\":\"auth_key\",\"collection\":\"libraryItem\",\"changes\":[{\"_id\":\"id\",\"removed\":false,\"temp\":false,\"_ctime\":\"2020-01-01T00:00:00Z\",\"_mtime\":\"2020-01-01T00:00:00Z\",\"state\":{\"lastWatched\":null,\"timeWatched\":0,\"timeOffset\":0,\"overallTimeWatched\":0,\"timesWatched\":0,\"flaggedWatched\":0,\"duration\":0,\"video_id\":null,\"watched\":null,\"lastVidReleased\":null,\"noNotif\":false},\"name\":\"name\",\"type\":\"type_name\",\"poster\":null,\"behaviorHints\":{\"defaultVideoId\":null}}]}" =>
+            {
+                Box::new(future::ok(Box::new(APIResult::Ok {
+                    result: SuccessResponse { success: True {} },
+                }) as Box<dyn Any>))
+            }
+            _ => default_fetch_handler(request),
+        }
+    }
+    let meta_preview = MetaPreview {
+        id: "id".to_owned(),
+        type_name: "type_name".to_owned(),
+        name: "name".to_owned(),
+        poster: None,
+        logo: None,
+        description: None,
+        release_info: None,
+        runtime: None,
+        released: None,
+        poster_shape: Default::default(),
+        trailer: None,
+        behavior_hints: Default::default(),
+    };
+    let lib_item = LibItem {
+        id: "id".to_owned(),
+        removed: false,
+        temp: false,
+        ctime: Some(Utc.ymd(2020, 1, 1).and_hms_milli(0, 0, 0, 0)),
```

Comment:
the concept of this test is to prove that ctime and state wont be updated but mtime will be.